CMakeLists.txt

```diff
@@ -532,6 +530,57 @@ while(library_variants_copy)
             ${INSTALL_DIR}/lib/${target} ${install_directory}/lib
     )
 
+    #picolibc tests
+    string(REPLACE " " ";" clang_compile_flags ${runtimes_flags})
+    add_custom_target(check-picolibc_${variant})
+    set(picolibc_test_ld ${picolibc_SOURCE_DIR}/scripts/test-arm.ld)
+    if(${variant} EQUAL "aarch64")
+        set(picolibc_test_ld ${picolibc_SOURCE_DIR}/scripts/test-aarch64.ld)
+    endif()
+    set(picolibc_tests printf_scanf
+                        printf-tests
+                        rand
+                        regex
+                        ungetc
+                        fenv
+                        long_double
+                        malloc
+                        ffs
+                        setjmp
+                        atexit
+                        on_exit
+                        math-funcs
+                        timegm
+                        time-tests
+                        test-strtod
+                        test-strchr
+                        test-memset
+                        test-put
+                        test-efcvt
+                        malloc_stress
+                        posix-io)
+    foreach(test IN LISTS picolibc_tests)
+        add_custom_target(
```

Comment:
Use add_custom_command here. add_custom_target is always considered out of date: https://cmake.org/cmake/help/latest/command/add_custom_target.html